lib/Gateway.js

```diff
@@ -1631,82 +1631,78 @@ Gateway.prototype.discoverValue = function (node, vId) {
-                break
-              default:
-                cfg.discovery_payload.device_class = 'None'
-            }
-            cfg = setBinaryPayloadFromSensor(cfg, valueId, off)
-          } else {
-            // https://github.com/zwave-js/node-zwave-js/blob/master/packages/zwave-js/src/lib/commandclass/NotificationCC.ts
-            // https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/notifications.json
-            cfg = copy(hassCfg.sensor_generic)
-            cfg.object_id = utils.joinProps(
-              'notification',
-              valueId.property,
-              valueId.propertyKey
-            )
-            // TODO: Improve the icons for different propertyKeys!
-            switch (valueId.propertyKey) {
-              case 'Motion sensor status':
-                cfg.discovery_payload.icon = 'mdi:motion-sensor'
-                break
-              default:
-                cfg.discovery_payload.icon = 'mdi:alarm-light'
-            }
-            cfg.discovery_payload.value_template = getMappedStateTemplate(
-              valueId.states,
-              valueId.default
-            )
+        if (valueId.list && valueId.states.length === 2) {
+          let off = 0 // set default off to 0.
+          cfg = copy(hassCfg.binary_sensor) // Setup the configuration using default values
+          cfg.object_id = valueId.propertyKey
+          switch (valueId.propertyKeyName) {
+            case 'Access Control':
+              cfg.discovery_payload.device_class = 'lock'
+              off = 23 // Closed state
+              break
+            case 'Cover status':
+              cfg.discovery_payload.device_class = 'opening'
+              break
+            case 'Door state':
+              cfg.discovery_payload.device_class = 'door'
+              off = 1 // Door closed on payload 1
+              break
+            case 'Dust in device status':
+            case 'Over-current status':
+            case 'Over-load status':
+            case 'Hardware status':
+              cfg.discovery_payload.device_class = 'problem'
+              break
+            case 'Heat sensor status':
+              cfg.discovery_payload.device_class = 'heat'
+              break
+            case 'Motion sensor status':
+              cfg.discovery_payload.device_class = 'motion'
+              break
+            case 'Alarm status':
+              cfg.discovery_payload.device_class = 'connectivity'
+              break
+            // sensor status has multiple Properties. therefore is good to work
+            // on property basis... user friendly
+            case 'Sensor status':
+              switch (valueId.propertyName) {
+                case 'Smoke Alarm':
+                  cfg.discovery_payload.device_class = 'smoke'
+                  break
+                case 'Water Alarm':
+                  cfg.discovery_payload.device_class = 'moisture'
+                  break
+                default:
```

Comment:
same here i miss `None` class.

Reply:
You don't need it because None is setup as the default up above for binary_sensor.